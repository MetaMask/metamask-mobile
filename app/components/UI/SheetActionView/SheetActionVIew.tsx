import React from 'react';
import { View } from 'react-native';
import {
  ButtonSize,
  ButtonVariants,
} from '../../../component-library/components/Buttons/Button';
import Button from '../../../component-library/components/Buttons/Button/Button';
import { strings } from '../../../../locales/i18n';
import createStyles from './SheetActionView.styles';
import { SheetActionViewI } from './SheetActionView.types';

const SheetActionView = ({ onConfirm, onCancel }: SheetActionViewI) => {
  const styles = createStyles();
  return (
    <View style={styles.actionsContainer}>
      <Button
        label={strings('action_view.cancel')}
        onPress={onCancel}
        variant={ButtonVariants.Secondary}
        size={ButtonSize.Lg}
        style={styles.cancelButton}
      />
      <Button
        label={strings('action_view.confirm')}
        onPress={onConfirm}
        variant={ButtonVariants.Primary}
        size={ButtonSize.Lg}
        style={styles.confirmButton}
      />
    </View>
  );
};
export default SheetActionView;
