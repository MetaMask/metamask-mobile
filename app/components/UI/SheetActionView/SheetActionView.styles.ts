import { StyleSheet } from 'react-native';

const createStyles = () =>
  StyleSheet.create({
    actionsContainer: {
      flexDirection: 'row',
      padding: 16,
    },
    cancelButton: { flex: 1, marginRight: 8 },
    confirmButton: { flex: 1, marginLeft: 8 },
  });

export default createStyles;
