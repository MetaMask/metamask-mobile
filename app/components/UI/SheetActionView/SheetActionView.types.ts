export interface SheetActionViewI {
  onCancel: () => void;
  onConfirm: () => void;
}
