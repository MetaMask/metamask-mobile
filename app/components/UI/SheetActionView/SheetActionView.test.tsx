import React from 'react';
import renderWithProvider from '../../../util/test/renderWithProvider';
import SheetActionView from './SheetActionVIew';

describe('SheetActionView', () => {
  it('should render correctly', () => {
    const { toJSON } = renderWithProvider(
      <SheetActionView onCancel={() => null} onConfirm={() => null} />,
      { state: {} },
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
