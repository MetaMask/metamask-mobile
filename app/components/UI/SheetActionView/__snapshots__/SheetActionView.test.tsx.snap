// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SheetActionView should render correctly 1`] = `
<View
  style={
    {
      "flexDirection": "row",
      "padding": 16,
    }
  }
>
  <TouchableOpacity
    accessibilityRole="button"
    accessible={true}
    activeOpacity={1}
    onPress={[Function]}
    onPressIn={[Function]}
    onPressOut={[Function]}
    style={
      {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "backgroundColor": "transparent",
        "borderColor": "#0376C9",
        "borderRadius": 24,
        "borderWidth": 1,
        "flex": 1,
        "flexDirection": "row",
        "height": 48,
        "justifyContent": "center",
        "marginRight": 8,
        "paddingHorizontal": 16,
      }
    }
    variant="Secondary"
  >
    <Text
      accessibilityRole="text"
      style={
        {
          "color": "#0376C9",
          "fontFamily": "Euclid Circular B",
          "fontSize": 14,
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 22,
        }
      }
    >
      Cancel
    </Text>
  </TouchableOpacity>
  <TouchableOpacity
    accessibilityRole="button"
    accessible={true}
    activeOpacity={1}
    onPress={[Function]}
    onPressIn={[Function]}
    onPressOut={[Function]}
    style={
      {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "backgroundColor": "#0376C9",
        "borderRadius": 24,
        "flex": 1,
        "flexDirection": "row",
        "height": 48,
        "justifyContent": "center",
        "marginLeft": 8,
        "paddingHorizontal": 16,
      }
    }
    variant="Primary"
  >
    <Text
      accessibilityRole="text"
      style={
        {
          "color": "#FFFFFF",
          "fontFamily": "Euclid Circular B",
          "fontSize": 14,
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 22,
        }
      }
    >
      Confirm
    </Text>
  </TouchableOpacity>
</View>
`;
